src/rebar3_hex_app.erl

```diff
@@ -0,0 +1,205 @@
+%%% ===================================================================
+%%% App utils
+%%% ===================================================================
+-module(rebar3_hex_app).
+
+-export([vcs_vsn/2, find/2, get_deps/1, validate/1]).
+
+-define(VALIDATIONS, [
+    has_semver,
+    has_contributors,
+    has_maintainers,
+    has_description,
+    has_licenses,
+    has_unstable_deps
+]).
+
+-type app_data() :: #{
+    name := string(),
+    version := string(),
+    details := list(),
+    deps := list()
+}.
+
+-type validation_errors() :: #{
+    warnings => list(),
+    errors => list()
+}.
+
+-spec find([rebar_app_info:t()], string()) -> {ok, rebar_app_info:t()} | {error, app_not_found}.
+find(Apps, Name) ->
```

Comment:
I believe `rebar_app_utils` has a similar function, we could just rip this out and use it. No strong feelings.